packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -347,28 +413,129 @@ async function selectPlatformAndExecute(
     });
 
     if (targetName === undefined) {
-      vscode.window.showInformationMessage(commandCancelledMessage);
-      return;
+      // user cancelled operation
+      return undefined;
     }
   }
 
-  await startServer(false, componentName, startTime);
-
-  // New target device entered
-  if (targetName !== '') {
+  // new target device entered
+  if (targetName && targetName !== '') {
     PreviewService.instance.updateRememberedDevice(
       platformSelection.platformName,
       targetName
     );
+
     target = targetName;
   }
 
-  const previewCommand = new SfdxCommandBuilder()
+  return target;
+}
+
+/**
+ * Prompts the user to select an app to preview the LWC on. Defaults to browser.
+ *
+ * @param platformSelection the selected platform
+ * @param configFile path to a config file
+ * @returns the name of the selected device or undefined if user cancels selection.
+ */
+async function selectTargetApp(
+  platformSelection: PreviewQuickPickItem,
+  configFile: string | undefined
+): Promise<string | undefined> {
+  let targetApp: string | undefined = 'browser';
+  const options: vscode.QuickPickItem[] = [];
+
+  if (configFile === undefined || fs.existsSync(configFile) === false) {
+    return targetApp;
+  }
+
+  try {
+    const fileContent = fs.readFileSync(configFile, 'utf8');
+    const json = JSON.parse(fileContent);
+    const appDefinitionsForSelectedPlatform =
```

Comment:
Not necessarily in this PR, but we need to look at how we can expose the underlying functionality we have in the `lwc-dev-mobile` plugin to make management of the app configuration more robust here, like we do there.

@xyc maybe you know, is there a standard mechanism plugin developers use to share code that isn't otherwise exposed through the plugin interface, between their plugins and the Extension? My first thought is to have a shared Node module dependency between the two, e.g. `@salesforce/lwc-dev-mobile-core`, but I'm curious if there's another way.

Reply:
@khawkins Yes. In fact, Node module dependencies are the directions we are moving towards to replace CLI commands. We have and are moving a lot of our dependencies on SFDX CLI to node modules, and make the extension solely dependent on node modules. This has brought us these benefits:
- Users don't have to install CLI and additional plugins. Quite a few users have configuration issues while installing CLI
- Significant performance improvement in commands. There is a heavy overhead when oclif looks up plugins.

Ideally if the mobile preview relies on a node module and not CLI plugin that would be amazing, but if part of the functionalities are moved to node module that's fine by me too.